base-template.sh

```diff
@@ -89,8 +91,10 @@ load_install_dir() {
 # Returns: None
 #####################################################
 set_main_variables() {
-    HOOK_NAME=$(basename "$0")
-    HOOK_FOLDER=$(dirname "$0")
+
+    HOOK_NAME="$GITHOOKS_HOOK_NAME"
```

Comment:
This could really use a comment stating these are set by the wrapper script, IMHO.